File: my_cpp_pkg/src/smartphone.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SmrtphoneNode : public rclcpp::Node 
{
public:
    SmrtphoneNode() : Node("smartphone"){
        subscriber_ = this->create_subscription<example_interfaces::msg::String>(
            "robot_news", 10,
        std::bind(&SmrtphoneNode::callbackRobotNews, this, std::placeholders::_1)); // gdyby były dwa, to w ted std::placeholders::_2, etc

        RCLCPP_INFO(this->get_logger(), "Smrtphone je has been started");
    }

private:
    void callbackRobotNews(const example_interfaces::msg::String::SharedPtr msg){
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }

    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SmrtphoneNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: my_py_pkg/my_py_pkg/smartphone.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String

class SmartphoneNode(Node):
    def __init__(self):
        super().__init__("smartphone")

        self.subscriber = self.create_subscription(String, "robot_news",
                self.callback_robot_news, 10) # topic type, topic_name, callback, queue

        self.get_logger().info(f"{self.__class__.__name__} has been started")

    def callback_robot_news(self, msg):  # callback_[name_of_topic] 
        self.get_logger().info(msg.data)  # bo w String, jest pole data


def main(args=None):
    rclpy.init(args=args)
    node = SmartphoneNode()
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()